### PR TITLE
Supermatter Autodoc + Code Cleanup

### PR DESCRIPTION
**Pull request by redmoogle**
### Intent of your Pull Request

Cleans the code up
Also adds autodoc for variables

### Why is this good for the game?

Clean code is good

#### Changelog

:cl:  
tweak: SM has cleaner code
/:cl:

### CODE REVIEW

File: code/modules/power/supermatter/supermatter.dm [hunk 1]
```diff
@@ -13,9 +18,9 @@
 #define NITROGEN_HEAT_PENALTY -1.5
 #define H2O_HEAT_PENALTY 12 //This'll get made slowly over time, I want my spice rock spicy god damnit
 
-//All of these get divided by 10-bzcomp * 5 before having 1 added and being multiplied with power to determine rads
-//Keep the negative values here above -10 and we won't get negative rads
-#define OXYGEN_TRANSMIT_MODIFIER 1.5   //Higher == Bigger bonus to power generation.
+// Higher == Bigger bonus to power generation.
```
Comment (Bibby0110) [inline code/modules/power/supermatter/supermatter.dm]:
```suggestion
/// Higher == Bigger bonus to power generation. 
```

File: code/modules/power/supermatter/supermatter.dm [hunk 2]
```diff
@@ -2,9 +2,14 @@
...
 //Modifications include removing the world-ending full supermatter variation, and leaving only the shard.
 
+// How much heat before SM prints warnings
+#define CRITICAL_TEMPERATURE 10000
+
+// How much energy before SM prints tesla warnings
 #define SUPERMATTER_MAXIMUM_ENERGY 1e6
 
-#define PLASMA_HEAT_PENALTY 15     // Higher == Bigger heat and waste penalty from having the crystal surrounded by this gas. Negative numbers reduce penalty.
+// Higher == Bigger heat and waste penalty from having the crystal surrounded by this gas. Negative numbers reduce penalty.
```
Comment (Bibby0110) [inline code/modules/power/supermatter/supermatter.dm]:
```suggestion
/// Higher == Bigger heat and waste penalty from having the crystal surrounded by this gas. Negative numbers reduce penalty.
```

File: code/modules/power/supermatter/supermatter.dm [hunk 3]
```diff
@@ -2,9 +2,14 @@
 //Please do not bother them with bugs from this port, however, as it has been modified quite a bit.
 //Modifications include removing the world-ending full supermatter variation, and leaving only the shard.
 
+// How much heat before SM prints warnings
+#define CRITICAL_TEMPERATURE 10000
+
+// How much energy before SM prints tesla warnings
```
Comment (Bibby0110) [inline code/modules/power/supermatter/supermatter.dm]:
```suggestion
/// How much energy before SM prints tesla warnings
```

File: code/modules/power/supermatter/supermatter.dm [hunk 4]
```diff
@@ -2,9 +2,14 @@
 //Please do not bother them with bugs from this port, however, as it has been modified quite a bit.
 //Modifications include removing the world-ending full supermatter variation, and leaving only the shard.
 
+// How much heat before SM prints warnings
```
Comment (Bibby0110) [inline code/modules/power/supermatter/supermatter.dm]:
```suggestion
/// How much heat before SM prints warnings
```

File: code/modules/power/supermatter/supermatter.dm [hunk 5]
```diff
@@ -13,9 +18,9 @@
 #define NITROGEN_HEAT_PENALTY -1.5
 #define H2O_HEAT_PENALTY 12 //This'll get made slowly over time, I want my spice rock spicy god damnit
 
-//All of these get divided by 10-bzcomp * 5 before having 1 added and being multiplied with power to determine rads
-//Keep the negative values here above -10 and we won't get negative rads
-#define OXYGEN_TRANSMIT_MODIFIER 1.5   //Higher == Bigger bonus to power generation.
+// Higher == Bigger bonus to power generation. 
+// All of these get divided by 10-bzcomp * 5 before having 1 added and being multiplied with power to determine rads
```
Comment (Bibby0110) [inline code/modules/power/supermatter/supermatter.dm]:
```suggestion
/// All of these get divided by 10-bzcomp * 5 before having 1 added and being multiplied with power to determine rads
```

File: code/modules/power/supermatter/supermatter.dm [hunk 6]
```diff
@@ -25,9 +30,11 @@
 #define HEALIUM_TRANSMIT_MODIFIER 2.4
 #define PLUONIUM_TRANSMIT_MODIFIER 15
 
-#define BZ_RADIOACTIVITY_MODIFIER 5 //Improves the effect of transmit modifiers
+// How much extra radioactivity to emit
```
Comment (Bibby0110) [inline code/modules/power/supermatter/supermatter.dm]:
```suggestion
/// How much extra radioactivity to emit
```

File: code/modules/power/supermatter/supermatter.dm [hunk 7]
```diff
@@ -25,9 +30,11 @@
...
+#define BZ_RADIOACTIVITY_MODIFIER 5 // Up to 500% rads
 
-#define N2O_HEAT_RESISTANCE 6          //Higher == Gas makes the crystal more resistant against heat damage.
+//Higher == Gas makes the crystal more resistant against heat damage.
```
Comment (Bibby0110) [inline code/modules/power/supermatter/supermatter.dm]:
```suggestion
/// Higher == Gas makes the crystal more resistant against heat damage.
```

File: code/modules/power/supermatter/supermatter.dm [hunk 8]
```diff
@@ -85,95 +92,109 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
...
+	/// Variables for calculating SM Health
 	var/warning_point = 50
-	var/warning_alert = "Danger! Crystal hyperstructure integrity faltering!"
 	var/damage_penalty_point = 550
 	var/emergency_point = 700
-	var/emergency_alert = "CRYSTAL DELAMINATION IMMINENT."
 	var/explosion_point = 900
 
-	var/emergency_issued = FALSE
+	/// Strings for radio
```
Comment (Bibby0110) [inline code/modules/power/supermatter/supermatter.dm]: Don't autodoc general comments, autodoc for individual variables

File: code/modules/power/supermatter/supermatter.dm [hunk 9]
```diff
@@ -85,95 +92,109 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
...
-
+	/// This is set to true when the SM is about to explode and has the blue shields up
 	var/final_countdown = FALSE
 
+	/// Health of the SM
 	var/damage = 0
 	var/damage_archived = 0
-	var/safe_alert = "Crystalline hyperstructure returning to safe operating parameters."
+
+	/// Variables for calculating SM Health
```
Comment (Bibby0110) [inline code/modules/power/supermatter/supermatter.dm]: Don't autodoc general comments, autodoc for individual variables

File: code/modules/power/supermatter/supermatter.dm [hunk 10]
```diff
@@ -85,95 +92,109 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
...
-//	var/config_power_reduction_per_tick = 0.5
-	//How much hallucination should it produce per unit of power?
+
+	///How much hallucination should it produce per unit of power?
 	var/config_hallucination_power = 0.1
 
-	var/support_integrity = 100 //integrity of the support base, used when antinoblium is attached
-	var/antinoblium_attached = FALSE
-	var/corruptor_attached = FALSE
+	/// Antinobilium related variables
```
Comment (Bibby0110) [inline code/modules/power/supermatter/supermatter.dm]: Don't autodoc general comments, autodoc for individual variables

File: code/modules/power/supermatter/supermatter.dm [hunk 11]
```diff
@@ -85,95 +92,109 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
...
+	var/support_integrity = 100			// Current support integrity. Provides *fun* effects
+	var/antinoblium_attached = FALSE	// The thing that makes it explode more
+	var/corruptor_attached = FALSE		// The thing that reduces support integrity
 
+	/// Radio related variables
```
Comment (Bibby0110) [inline code/modules/power/supermatter/supermatter.dm]: Don't autodoc general comments, autodoc for individual variables

File: code/modules/power/supermatter/supermatter.dm [hunk 12]
```diff
@@ -85,95 +92,109 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
...
 	var/obj/item/radio/radio
 	var/radio_key = /obj/item/encryptionkey/headset_eng
-	var/engineering_channel = "Engineering"
+	var/engineering_channel = RADIO_CHANNEL_ENGINEERING
 	var/common_channel = null
 
-	//for logging
+	/// Logging
```
Comment (Bibby0110) [inline code/modules/power/supermatter/supermatter.dm]: Don't autodoc general comments, autodoc for individual variables

File: code/modules/power/supermatter/supermatter.dm [hunk 13]
```diff
@@ -85,95 +92,109 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
...
 	var/has_been_powered = FALSE
 	var/has_reached_emergency = FALSE
 
-	// For making hugbox supermatter
-	var/takes_damage = TRUE
-	var/produces_gas = TRUE
+	/// Hugbox Supermatter Variables
```
Comment (Bibby0110) [inline code/modules/power/supermatter/supermatter.dm]: Don't autodoc general comments, autodoc for individual variables

File: code/modules/power/supermatter/supermatter.dm [hunk 14]
```diff
@@ -85,95 +92,109 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
...
+	var/takes_damage = TRUE		// Does the SM take integrity damage
+	var/produces_gas = TRUE		// Does it make gas
+
+	/// Holder variables
```
Comment (Bibby0110) [inline code/modules/power/supermatter/supermatter.dm]: Don't autodoc general comments, autodoc for individual variables

**Review by Bibby0110 — CHANGES_REQUESTED** [review]
holy moly SM code was bad

**Review by JamieD1 — APPROVED** [review]

**Pull request merged**